### PR TITLE
refactor: update code challenge boilerplate

### PR DESCRIPTION
**Pull request by talo242**
# What this PR does

This PR adds new updates to the initial boilerplate used on our code challenge:

### General changes:
- The PR adds Prettier to both the backend & frontend directories to have automatic code formatting
- Moves the `.env` file to the root of the code challenge project to make it easier to run docker commands

### Backend changes:
- Adds a basic CORS configuration to make sure the code challenge will work out-of-the-box
- Adds the database migration and seed to the initial docker command
When candidates run `docker compose up --build` this will happen:
  1. The database image will be fetched from the internet
  2. The frontend image will be created
  3. The backend image will be created
  4. The database migrations will run
  5.  The seed script will run and the database will be populated with dummy data

### Frontend changes:
- Adds a basic `Policies` feature that fetches and renders a list of policies
image
- Refactors the directory structure to make it more similar to our real projects
Now we have a `components` and `features` folders, and we are following a module architecture:
```
| features
| - Policies
|    | - index.tsx
|    | - Policies.tsx
|    | - Policies.model.ts
|    | - Policies.test.ts
|    | - Policies.mocks.ts
```
- Adds [MSW](https://mswjs.io/) to the project to mock API calls
- Adds some test examples where added to show candidates what is the expected test patterns

### CODE REVIEW

**Review by killthekitten — APPROVED** [review]
I really like how you restructured, maybe even too pretty for a coding challenge 😅 Kudos for fixing the Readme/docker part and the tests!

File: frontend/src/features/Policies/Policies.model.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+type InsuranceType = 'HEALTH' | 'LIABILITY' | 'HOUSEHOLD';
+
+export type PolicyStatus = 'ACTIVE' | 'PENDING' | 'CANCELLED' | 'DROPPED_OUT';
+
+export type Provider = 'BARMER' | 'PENDING' | 'CANCELLED' | 'DROPPED_OUT';
```
Comment (kseha1080) [inline frontend/src/features/Policies/Policies.model.ts:5]: This one needs an update

File: frontend/src/features/Policies/Policies.model.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  id: string;
+  lastName: string;
+}
+
+export interface Policy {
+  customer: Customer;
+  endDate: Date | null;
+  id: string;
+  insuranceType: InsuranceType;
+  provider: 'BARMER';
```
Comment (kseha1080) [inline frontend/src/features/Policies/Policies.model.ts:19]: Also here

**Review by kseha1080 — CHANGES_REQUESTED** [review]
Thanks a lot for the update on this 👍 I do think that this much boilerplate might be a bit too much to give to the candidates, such as TS types all laid out for them to use. What we want to do is give them the least amount of boilerplate + config to start building but still need to see how they handle types, logic, etc. right? wdyt?

**Review by kseha1080 — APPROVED** [review]

**Pull request merged**